plugins/pulp_rpm/plugins/importers/yum/sync.py

```diff
@@ -77,12 +81,49 @@ def set_progress(self):
     @property
     def sync_feed(self):
         """
-        :return:    the URL of the feed we should sync
-        :rtype:     str
+        :return:    list the URLs of the feed we should sync
+        :rtype:     list
         """
         repo_url = self.call_config.get(importer_constants.KEY_FEED)
-        if repo_url is not None and not repo_url.endswith('/'):
-            repo_url += '/'
+        if repo_url:
+            repo_url_slash = repo_url
+            self.tmp_dir = tempfile.mkdtemp(dir=self.working_dir)
+            if not repo_url.endswith('/'):
+                    repo_url_slash = repo_url + '/'
+            try:
+                self.check_metadata(repo_url_slash)
+                return [repo_url_slash]
+            except:
+                # treat as mirrorlist
+                return self.parse_as_mirrorlist(repo_url)
+            finally:
+                shutil.rmtree(self.tmp_dir, ignore_errors=True)
+        return [repo_url]
+
+    def parse_as_mirrorlist(self, feed):
+        """
```

Comment:
Add a description here for what this method does. Also, if this method is not intended to be used outside of this class, you might rename it to ```_parse_as_mirrorlist()```.